File: Swift/Sources/App/Models/Task.swift
import Fluent
import Vapor

final class Task: Model, Content {
    static let schema = "tasks"

    @ID(key: .id)
    var id: UUID?

    @Parent(key: "user_id")
    var user: User

    @Field(key: "title")
    var title: String

    @Field(key: "description")
    var description: String

    @Field(key: "is_done")
    var isDone: Bool

    init() { }

    init(id: UUID? = nil, userID: User.IDValue, title: String, description: String, isDone: Bool = false) {
        self.id = id
        self.$user.id = userID
        self.title = title
        self.description = description
        self.isDone = isDone
    }
}

// DTOs
struct CreateTaskDTO: Content {
    let title: String
    let description: String
}

struct UpdateTaskDTO: Content {
    let title: String?
    let description: String?
    let is_done: Bool?
}

File: Elixir/lib/magnetar/todos/task.ex
defmodule Magnetar.Todos.Task do
  use Ecto.Schema
  import Ecto.Changeset

  @derive {Jason.Encoder, only: [:id, :title, :description, :is_done, :user_id]}
  schema "tasks" do
    field :title, :string
    field :description, :string
    field :is_done, :boolean, default: false
    field :user_id, :id

    timestamps()
  end

  @doc false
  def changeset(task, attrs) do
    task
    |> cast(attrs, [:title, :description, :is_done, :user_id])
    |> validate_required([:title, :description, :user_id])
    |> foreign_key_constraint(:user_id)
  end
end
